src/Process.php

```diff
@@ -120,11 +122,15 @@ public function start(LoopInterface $loop, $interval = 0.1)
             });
         };
 
-        $this->stdin  = new Stream($this->pipes[0], $loop);
-        $this->stdin->pause();
-        $this->stdout = new Stream($this->pipes[1], $loop);
+        $this->stdin  = new WritableResourceStream($this->pipes[0], $loop);
+        $this->stdin->on('close', function () {
+            if (is_resource($this->pipes[0])) {
+                fclose($this->pipes[0]);
```

Comment:
Currently this is needed until https://github.com/reactphp/stream/pull/107 is merged and tagged